### ISSUE TITLE
Latest tag on dockerhub points to snapshot instead of latest stable

### ISSUE THREAD

**Issue by mulder999** [DependencyTrack/dependency-track#1644]
### Current Behavior:

On registry dockerhub registry:
`dependencytrack/apiserver:latest` image has digest `4d9653d477fe` which corresponds to `dependencytrack/apiserver:snapshot` image

[Reference](https://hub.docker.com/r/dependencytrack/apiserver/tags)

### Expected Behavior:

`dependencytrack/apiserver:latest` should point to latest stable version, namely `dependencytrack/apiserver:4.5.0` as of today

**Comment by stevespringett** [comment]
@nscuro @k3rnelpan1c-dev 

This isn't good and is partially my doing. The `:latest` tag was not pushed on release, so I modified [_meta-build.yaml#L92](https://github.com/DependencyTrack/dependency-track/blob/master/.github/workflows/_meta-build.yaml#L92) and added it. However, it appears that this block of code is also responsible for pushing on CI, in which case we don't want the latest tag being pushed.

Two questions:
1) How do we correct this so that CI builds do not push latest?
2) How do we correct Hub so that `latest` is the same as `4.5.0`?

**Comment by k3rnelpan1c-dev** [comment]
@stevespringett 

currently you will have to re-tag and push `latest` manually on your machine.
sorry for missing the latest tag logic in the original CI, but your changes cause the latest tag to follow the edge development on the default branch.

I will be doing more CI work this weekend / later today, so I will get on fixing this ASAP

**Comment by k3rnelpan1c-dev** [comment]
the only issue with re-taging mannualy is that this will be platform specific :thinking:

**Comment by nscuro** [comment]
> the only issue with re-taging manually is that this will be platform specific 🤔

It looks like `docker buildx imagetools create` can do this, as it's just working on manifests and not the images themselves: https://docs.docker.com/engine/reference/commandline/buildx_imagetools_create/#usage

```shell
docker buildx imagetools create dependencytrack/apiserver:4.5.0 --tag dependencytrack/apiserver:latest
```

Here's what it would push to the registry:

```shell
$ docker buildx imagetools create --dry-run dependencytrack/apiserver:4.5.0
{
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "schemaVersion": 2,
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "digest": "sha256:9365f306ac54eaf1216ad8f2846062b3fe538399d8fb10a11e82be20c8c8e797",
         "size": 1369,
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "digest": "sha256:4d9b0677a2c4b9f8d28fd2132a0125c3d327014058be57bb6ad758b56762cac6",
         "size": 1369,
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

**Comment by nscuro** [comment]
For testing purposes I created the `dependencytrack/apiserver:issue-1644` tag from the current `dependencytrack/apiserver:snapshot` tag:

```shell
docker buildx imagetools create --tag dependencytrack/apiserver:issue-1644 dependencytrack/apiserver:snapshot
```

and it worked as expected: https://hub.docker.com/layers/217931103/dependencytrack/apiserver/issue-1644/images/sha256-63c092bc4be6720199a68f864fc7fd4cb71ea2a1b3097426b9fa1db7a732df06?context=repo

Both platfoms are included and the image digests match.

**Comment by k3rnelpan1c-dev** [comment]
Just checked and yes a `skopeo copy -a docker://docker.io/dependencytrack/apiserver:4.5.0 docker://docker.io/dependencytrack/apiserver:latest` would work too

**Comment by nscuro** [comment]
`latest` tags have been corrected to reference the respective `4.5.0` images again.
